x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageOwnApiKeyClusterPrivilege.java

```diff
@@ -60,7 +60,7 @@ protected boolean extendedCheck(String action, TransportRequest request, Authent
                     invalidateApiKeyRequest.getUserName(), invalidateApiKeyRequest.getRealmName(),
                     invalidateApiKeyRequest.ownedByAuthenticatedUser());
             }
-            return false;
+            throw new IllegalArgumentException("manage own api key privilege only supports API key requests");
```

Comment:
Can we put the request class into the exception?
```suggestion
            throw new IllegalArgumentException("manage own api key privilege only supports API key requests (not " + request.getClass().getName() + ")");
```